packages/api/src/external/hie/__tests__/set-doc-query-progress.test.ts

```diff
@@ -334,4 +344,315 @@ describe("aggregateAndSetHIEProgresses", () => {
       requestId,
     });
   });
+
+  it("has download completed when overallDocQueryProgress has download processing, externalData is empty", async () => {
+    const externalData: PatientExternalData = {};
+
+    const overallDocQueryProgress: DocumentQueryProgress = {
+      ...docQueryProgress,
+      convert: undefined,
+    };
+
+    const aggregateAndSetHIEProgressesResult = aggregateAndSetHIEProgresses(
+      overallDocQueryProgress,
+      externalData
+    );
+
+    expect(aggregateAndSetHIEProgressesResult).toEqual({
+      download: { total: 0, errors: 0, status: "completed", successful: 0 },
+      requestId,
+    });
+  });
+});
```

Comment:
So yea the purpose of this this is so when we have no `externalData` that we set the entire state to completed. Reasoning behind this test is because there is some logic in place where we reset the externalData and then the overall progress would be stuck in processing.